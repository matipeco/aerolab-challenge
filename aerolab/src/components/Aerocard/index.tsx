import { StyledAerocard } from "./style";
import Image from "next/image";

export const Aerocard = () => {
  return (
    <StyledAerocard>
      <div>
        <p className="aerocard-title">Aerocard</p>
        <Image
          src="/assets/icons/aeropay-2.svg"
          alt=""
          width="24"
          height="24"
        />
      </div>
      <div>
        <p>Matias Pecorale</p>
        <p>07/23</p>
      </div>
    </StyledAerocard>
  );
};
