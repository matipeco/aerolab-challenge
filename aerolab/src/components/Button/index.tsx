import { StyledButton } from "./style";
import type { FunctionComponent, PropsWithChildren } from "react";

type Props = PropsWithChildren<{
  className: string;
}>;

export const Button: FunctionComponent<Props> = ({ children, className }) => {
  return <StyledButton className={className}>{children}</StyledButton>;
};
