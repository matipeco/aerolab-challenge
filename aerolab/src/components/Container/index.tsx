import { StyledContainer } from "./style";
import type { FunctionComponent, PropsWithChildren } from "react";

export const Container: FunctionComponent<PropsWithChildren> = ({
  children,
}) => {
  return <StyledContainer>{children}</StyledContainer>;
};
