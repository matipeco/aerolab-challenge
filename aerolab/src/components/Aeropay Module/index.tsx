import { Aerocard } from "../Aerocard";
import { Button } from "../Button";
import { StyledAeropay } from "./style";
import Image from "next/image";
import { RadioButton } from "../RadioButton";

export const AeropayModule = () => {
  return (
    <StyledAeropay>
      <p className="module__title">Add Balance</p>
      <Aerocard />
      <div className="module__points">
        <RadioButton id="point-1000" label="1000" name="points" />
        <RadioButton id="point-5000" label="5000" name="points" />
        <RadioButton id="point-7500" label="7500" name="points" />
      </div>

      <Button className="module__button">
        <Image
          src="/assets/icons/aeropay-3.svg"
          alt=""
          width="24"
          height="24"
        />
        Add Points
      </Button>
    </StyledAeropay>
  );
};
