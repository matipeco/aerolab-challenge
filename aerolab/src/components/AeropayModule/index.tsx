import { Aerocard } from "../Aerocard";
import { Button } from "../Button";
import { StyledAeropay } from "./style";
import Image from "next/image";
import { RadioButton } from "../RadioButton";
import { ChangeEvent, useContext, useState } from "react";
import { Context } from "@/context";

export const AeropayModule = () => {
  const { setPoints } = useContext(Context)!;
  const [selectedPoints, setSelectedPoints] = useState(0);

  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
    const { value } = ev.target;
    setSelectedPoints(Number(value));
  };

  const handleClick = () => {
    setPoints((points) => {
      return points + selectedPoints;
    });
  };
  return (
    <StyledAeropay>
      <p className="module__title">Add Balance</p>
      <Aerocard />
      <div className="module__points">
        <RadioButton
          id="point-1000"
          label="1000"
          name="points"
          value="1000"
          onChange={handleChange}
        />
        <RadioButton
          id="point-5000"
          label="5000"
          name="points"
          value="5000"
          onChange={handleChange}
        />
        <RadioButton
          id="point-7500"
          label="7500"
          name="points"
          value="7500"
          onChange={handleChange}
        />
      </div>

      <Button className="module__button" onClick={handleClick}>
        <Image
          src="/assets/icons/aeropay-3.svg"
          alt=""
          width="24"
          height="24"
        />
        Add Points
      </Button>
    </StyledAeropay>
  );
};
