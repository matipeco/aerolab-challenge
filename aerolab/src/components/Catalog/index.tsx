import { Container } from "../Container";
import { StyledCatalog } from "./style";
import type { Product } from "@/pages";
import { ChangeEvent, FunctionComponent, useState } from "react";
import { ProductCard } from "../ProductCard";
import Image from "next/image";
import { RadioButton } from "../RadioButton";

type Props = {
  products: Product[];
};

export const Catalog: FunctionComponent<Props> = ({ products }) => {
  const categories = products.map((prod) => {
    return prod.category;
  });
  const setCategories = Array.from(new Set(categories));

  const [selectedFilter, setSelectedFilter] = useState("");

  const [selectedOrder, setSelectedOrder] = useState("most-recent");

  const catalogProducts = products
    .filter((prod) =>
      selectedFilter ? prod.category === selectedFilter : prod
    )
    .sort((a, b) => {
      if (selectedOrder === "most-recent") {
        return 0;
      }
      return selectedOrder === "highest-price"
        ? b.cost - a.cost
        : a.cost - b.cost;
    });

  const handleChange = (ev: ChangeEvent<HTMLSelectElement>) => {
    setSelectedFilter(ev.target.value);
  };

  const selectOrder = (ev: ChangeEvent<HTMLInputElement>) => {
    setSelectedOrder(ev.target.value);
  };

  return (
    <StyledCatalog>
      <Container>
        <h2 className="catalog__title" id="Catalog">
          <span>tech</span> products
        </h2>
        <div className="filters__container">
          <div className="select__container">
            <label htmlFor="filter">Filter by:</label>
            <select
              name=""
              id="filter"
              className="filters__select"
              onChange={handleChange}
            >
              <option value="">All Products</option>
              {setCategories.map((cat) => {
                return <option key={cat}>{cat}</option>;
              })}
            </select>
          </div>
          <div className="radio__container">
            <p>Sorted by:</p>
            <RadioButton
              id="radio-1"
              name="sort"
              label="Most recent"
              value="most-recent"
              onChange={selectOrder}
              checked={selectedOrder === "most-recent"}
            />
            <RadioButton
              id="radio-2"
              name="sort"
              label="Lowest Price"
              value="lowest-price"
              onChange={selectOrder}
              checked={selectedOrder === "lowest-price"}
            />
            <RadioButton
              id="radio-3"
              name="sort"
              label="Highest Price"
              value="highest-price"
              onChange={selectOrder}
              checked={selectedOrder === "highest-price"}
            />
          </div>

          <div className="pagination__container">
            <button>
              <Image
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
                className="chevron-left"
              />
            </button>

            <p>
              Page <span>1 of 2</span>
            </p>
            <button>
              <Image
                className="chevron-right"
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
              />
            </button>
          </div>
        </div>
        <div className="catalog__products">
          {catalogProducts.map((prod) => (
            <ProductCard
              image={prod.img.url}
              name={prod.name}
              key={prod._id}
              category={prod.category}
              cost={prod.cost}
              id={prod._id}
            />
          ))}
        </div>
        <div className="catalog__footer">
          <p>x of 32 products</p>
          <div className="pagination__container">
            <button>
              <Image
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
                className="chevron-left"
              />
            </button>

            <p>
              Page <span>1 of 2</span>
            </p>
            <button>
              <Image
                className="chevron-right"
                src="/assets/icons/chevron-default.svg"
                alt=""
                width="24"
                height="24"
              />
            </button>
          </div>
        </div>
      </Container>
    </StyledCatalog>
  );
};
