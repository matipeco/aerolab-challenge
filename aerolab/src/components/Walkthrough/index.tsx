import { Container } from "../Container";
import { WalkthroughCard } from "../WalkthroughCard";
import { StyledWalkthrough } from "./style";

export const Walkthrough = () => {
  return (
    <StyledWalkthrough>
      <Container>
        <WalkthroughCard
          image="/assets/illustrations/walkthroug-1-desktop.png"
          icon="/assets/icons/walkthrough-1.svg"
          title="1â€”browse"
          description="Browse our tech catalog with more than 20 top tech products"
        />
        <WalkthroughCard
          image="/assets/illustrations/walkthroug-2-desktop.png"
          icon="/assets/icons/walkthrough-2.svg"
          title="2â€”choose"
          description="Exchange your hard earned AeroPoints for the item you want"
        />
        <WalkthroughCard
          image="/assets/illustrations/walkthroug-3-desktop.png"
          icon="/assets/icons/walkthrough-3.svg"
          title="3â€”enjoy!"
          description="All done, you can relax! Weâ€™ll take care of delivery of your tech item!"
        />
      </Container>
    </StyledWalkthrough>
  );
};
