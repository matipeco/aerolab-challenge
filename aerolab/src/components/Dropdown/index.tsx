import { StyledDropdown, StyledContainer } from "./style";
import Image from "next/image";
import { useState, useContext } from "react";
import { AeropayModule } from "../AeropayModule";
import { Context } from "@/context";

export const Dropdown = () => {
  const [isExpanded, setIsExpanded] = useState(false);

  const { points } = useContext(Context)!;

  return (
    <StyledContainer>
      <StyledDropdown onClick={() => setIsExpanded(!isExpanded)}>
        <Image
          src="./assets/icons/aeropay-1.svg"
          alt=""
          width="32"
          height="32"
        />
        <span className="button__points">{points}</span>
        <Image
          src="./assets/icons/chevron-default.svg"
          alt=""
          width="24"
          height="24"
          className={`button__chevron ${isExpanded ? "expanded" : ""}`}
        />
      </StyledDropdown>

      {isExpanded && <AeropayModule />}
    </StyledContainer>
  );
};
