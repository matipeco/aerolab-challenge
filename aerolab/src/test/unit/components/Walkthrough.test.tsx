import { Walkthrough } from "@/components/Walkthrough";
import { render } from "@testing-library/react";

// jest.mock("../../../components/WalkthroughCard", () => {
//   return {
//     WalkthroughCard: () => "article",
//   };
// });
describe("Componente Walkthrough", () => {
  test("Se renderiza correctamente", () => {
    const { container } = render(<Walkthrough />);

    expect(container).toMatchSnapshot();
  });
});
