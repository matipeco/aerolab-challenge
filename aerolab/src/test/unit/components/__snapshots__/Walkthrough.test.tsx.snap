// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Walkthrough Se renderiza correctamente 1`] = `
<div>
  <section
    class="sc-51fa7f76-0 ivHqsm"
  >
    <div
      class="sc-50f7ce2-0 ipptrS"
    >
      <article
        class="sc-5d3fbbe4-0 dtykUg"
      >
        <img
          alt=""
          class="card__img"
          data-nimg="1"
          decoding="async"
          height="498"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-1-desktop.png&w=1080&q=75"
          srcset="/_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-1-desktop.png&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-1-desktop.png&w=1080&q=75 2x"
          style="color: transparent;"
          width="508"
        />
        <div
          class="card__text"
        >
          <p
            class="card__title"
          >
            <span>
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/assets/icons/walkthrough-1.svg"
                style="color: transparent;"
                width="40"
              />
            </span>
            1—browse
          </p>
          <p
            class="card__description"
          >
            Browse our tech catalog with more than 20 top tech products
          </p>
        </div>
      </article>
      <article
        class="sc-5d3fbbe4-0 dtykUg"
      >
        <img
          alt=""
          class="card__img"
          data-nimg="1"
          decoding="async"
          height="498"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-2-desktop.png&w=1080&q=75"
          srcset="/_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-2-desktop.png&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-2-desktop.png&w=1080&q=75 2x"
          style="color: transparent;"
          width="508"
        />
        <div
          class="card__text"
        >
          <p
            class="card__title"
          >
            <span>
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/assets/icons/walkthrough-2.svg"
                style="color: transparent;"
                width="40"
              />
            </span>
            2—choose
          </p>
          <p
            class="card__description"
          >
            Exchange your hard earned AeroPoints for the item you want
          </p>
        </div>
      </article>
      <article
        class="sc-5d3fbbe4-0 dtykUg"
      >
        <img
          alt=""
          class="card__img"
          data-nimg="1"
          decoding="async"
          height="498"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-3-desktop.png&w=1080&q=75"
          srcset="/_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-3-desktop.png&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fillustrations%2Fwalkthroug-3-desktop.png&w=1080&q=75 2x"
          style="color: transparent;"
          width="508"
        />
        <div
          class="card__text"
        >
          <p
            class="card__title"
          >
            <span>
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/assets/icons/walkthrough-3.svg"
                style="color: transparent;"
                width="40"
              />
            </span>
            3—enjoy!
          </p>
          <p
            class="card__description"
          >
            All done, you can relax! We’ll take care of delivery of your tech item!
          </p>
        </div>
      </article>
    </div>
  </section>
</div>
`;
